packages/transactions/src/__tests__/PaymentV2.spec.ts

```diff
@@ -115,14 +126,67 @@ describe('sign', () => {
     expect(deserializedPayment.signature).toEqual(signedPayment.signature)
   })
 
+  it("has the same signature when specifying token type 'hnt' as when omitting", async () => {
+    const { bob, alice } = await usersFixture()
+    const paymentWithoutTokenType = new PaymentV2({
+      payer: bob.address,
+      payments: [
+        {
+          payee: alice.address,
+          amount: 10,
+        },
+      ],
+      nonce: 1,
+    })
+
+    const paymentWithTokenType = new PaymentV2({
+      payer: bob.address,
+      payments: [
+        {
+          payee: alice.address,
+          amount: 10,
+        },
+      ],
+      nonce: 1,
+    })
+
+    const signedPaymentWithout = await paymentWithoutTokenType.sign({ payer: bob })
+    const signedPaymentWith = await paymentWithTokenType.sign({ payer: bob })
+    expect(signedPaymentWithout.toString()).toEqual(signedPaymentWith.toString())
+  })
+
   it('should sign the correct signature while handling an empty memo', async () => {
     const payment = await paymentFixture()
     payment.payments[0].memo = 'AAAAAAAAAAA='
     const reserializedTxn = PaymentV2.fromString(payment.toString())
     const { bob } = await usersFixture()
     const signedPayment = await reserializedTxn.sign({ payer: bob })
-    const decoded = proto.helium.blockchain_txn.decode(Buffer.from(signedPayment.toString(), 'base64'))
+    const decoded = proto.helium.blockchain_txn.decode(
+      Buffer.from(signedPayment.toString(), 'base64'),
+    )
     const base64Signature = (decoded.paymentV2?.signature as Buffer).toString('base64')
-    expect(base64Signature).toEqual('PiDvVpuSI0PViJi/ItQnqyMPbHFh3mmZSMpBzHuJ6TXlig1hDtdUE8is9FJBbhdJbSz9jCpHHBB2O2d0pYPSCw==')
+    expect(base64Signature).toEqual(
+      'TKUrksvBBTjgjoPEPhFc79+jPzaVN7VBWgKo4+GvDYZ4BO9HujpEuskMSvzVWq0T/tf2KAKd6d+fT3mYnMRYCA==',
+    )
+  })
+
+  it('should decode hnt token type to hnt', async () => {
+    const payment = await paymentFixture()
+    const { bob } = await usersFixture()
+    const signedPayment = await payment.sign({ payer: bob })
+    const decoded = proto.helium.blockchain_txn.decode(
+      Buffer.from(signedPayment.toString(), 'base64'),
+    )
+    expect(decoded.paymentV2?.payments?.[0].tokenType).toEqual(TokenType.hnt)
```

Comment:
When we create the payment `token_type` needs to be undefined, but when it's decoded it is 0. 🤔

Reply:
do we need the same kind of logic in the `fromString` method?

Reply:
hm I see this is calling the proto def directly